src/Server/GRPCServer.cpp

```diff
@@ -751,6 +752,35 @@ namespace
         session->authenticate(user, password, user_address);
         session->getClientInfo().quota_key = quota_key;
 
+        // Parse the OpenTelemetry traceparent header.
+        ClientInfo client_info = session->getClientInfo();
+        auto & client_metadata = responder->grpc_context.client_metadata();
+        auto traceparent = client_metadata.find("traceparent");
```

Comment:
Let's pass `traceparent` and `tracecontext` via QueryInfo's fields:

```
message QueryInfo {
   ...
   // comment
   string traceparent = 20;
   string tracestate = 21;
}
```

It would be more consistent because other options which are passed via QueryInfo's fields

Reply:
`traceparent` and `tracestate` are HTTP headers defined in https://www.w3.org/TR/trace-context/#traceparent-header. All existing OpenTelemetry instrumentation libraries that I know of that support W3C Trace Context propagation do so via HTTP headers/GRPC client metadata (which is HTTP headers underneath). Propagation this information via protobuf fields would break interoperability with OpenTelemetry libraries and require users to add manual instrumentation to inject the tracing context into the protobuf message. Do you think that is worth it?

In regards to consistency, this code was based off of the code that extracts these headers in HTTPHandler and was written to be consistent with that behavior, and changing it would introduce an inconsistency. Finally, `traceparent` and `tracestate` are, in my opinion, not really query info, and thus don't really belong in the QueryInfo message. I understand that this is more of an emotional argument than a logical one, though. Please let me know if my arguments to keeping these headers as headers makes sense, and if you still want me to change it.

Reply:
No, I just didn't know that passing those headers via GRPC client metadata is a kind of standard.